### PR TITLE
Make EventListener an Addressable

### LINKED ISSUE #39: Update EventListener to consume CloudEvent
# Expected Behavior

Once we have #36 the next step is for the running binary to expose a Knative addressable endpoint that can be invoked with a CloudEvent!

It doesn't have to actually do anything with the CloudEvent at this point but maybe could log something fun?

# Actual Behavior

Once we have #36 we'll have a running pod but it won't be listening for CloudEvents yet

# Additional Info

If we have #37 at this point, we should expand the end to end test to hit the running EventListener with a predefined CloudEvent. If not, we'll need to create a follow up issue in the #33 epic to add that to the end to end test.

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

An Addressable type needs to have an Address.Hostname/URL.
For EventListeners this should be the address of the EL Sink
service.

Addressable: https://github.com/knative/eventing/blob/master/docs/spec/interfaces.md
Fixes #39 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
The EventListener Status now contains an `Address.Hostname` and `Address.URL` 

```

### CODE REVIEW

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener.go [hunk 1]
```diff
@@ -188,6 +188,7 @@ func (c *Reconciler) reconcileService(el *v1alpha1.EventListener) error {
 			return err
 		} else {
 			el.Status.Configuration.Hostname = listenerHostname(service.Name, el.Namespace)
```
Comment (vtereso) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener.go]: We could remove this line I suppose and migrate the tests to this Addressable

Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener.go]: done!

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 2]
```diff
@@ -48,16 +50,15 @@ func init() {
...
-			),
+			bldr.EventListenerConfig(generatedResourceName),
 		),
 	)
 }
 
 var (
-	generatedResourceName string
+	generatedResourceName    string
+	ignoreLastTransitionTime = cmpopts.IgnoreTypes(apis.Condition{}.LastTransitionTime.Inner.Time)
```
Comment (vtereso) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: TIL

Comment (vtereso) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: Nice cleanup for sure

Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: :) I picked it up from the pipeline reconciler tests!

**Pull request merged**